app/controllers/ideas_controller.rb

```diff
@@ -9,20 +9,25 @@ def new
   def create
     @user = User.find(params[:user_id])
     @idea = @user.ideas.new(idea_params)
-      if @idea.save
-        redirect_to user_idea_path(@user, @idea)
-      else
-        render :new
-      end
+    if @idea.save
+      redirect_to user_idea_path(@user, @idea)
+    else
+      render :new
+    end
   end
 
   def index
     @ideas = Idea.all
   end
 
   def show
-    @idea = Idea.find(params[:id])
-    @user = @idea.user
+    # byebug
+    if current_user.id == params[:user_id].to_i
+      @idea = Idea.find(params[:id])
+      @user = @idea.user
+    else
+      render file: "/public/404"
```

Comment:
I'm feeling here that they will just hit that else if they are not the current_user.